From c55f0df912c7961ceafde9d8b3c873ff248ac77f Mon Sep 17 00:00:00 2001
From: Paulo Ewerton <pauloewerton@lsd.ufcg.edu.br>
Date: Tue, 15 Sep 2015 20:00:34 +0000
Subject: [PATCH] hz-page-header should allow string interpolations

As noted by Thai in his review for the patch
https://review.openstack.org/#/c/213879/, the hz-page-header
directive should be updated to allow string interpolations
such as {$ ... $} in order to support angular translate.

This patch can be tested by setting DISABLED = False in
_1051_project_ng_images_panel.py and _3031_identity_users_panel.py
and checking the header translation with the pseudo
translation tool.

Change-Id: If9778f361d2bb1682e258578b7525bb734b8f8b6
Closes-Bug: 1485778
---
 .../static/framework/widgets/headers/hz-page-header.directive.js  | 8 ++++----
 .../framework/widgets/headers/hz-page-header.directive.spec.js    | 6 ++----
 .../dashboards/identity/ngusers/templates/ngusers/index.html      | 4 ----
 .../identity/static/dashboard/identity/users/table/table.html     | 2 ++
 .../dashboards/project/ngimages/templates/ngimages/index.html     | 4 ----
 .../static/app/core/images/table/images-table.html                | 2 ++
 6 files changed, 10 insertions(+), 16 deletions(-)

diff --git a/horizon/static/framework/widgets/headers/hz-page-header.directive.js b/horizon/static/framework/widgets/headers/hz-page-header.directive.js
index 71b08ad..2b2b884 100644
--- a/horizon/static/framework/widgets/headers/hz-page-header.directive.js
+++ b/horizon/static/framework/widgets/headers/hz-page-header.directive.js
@@ -35,13 +35,13 @@
    * Default usage to provide a title header and a description that are
    * translated by the angular gettext translate filter.:
    *
-   * <hz-page-header header="'My Header' | translate" description="'foo' | translate"/>
+   * <hz-page-header header="{$ 'My Header' | translate $}" description="{$ 'foo' | translate $}"/>
    *
    * If you have additional content that you want to include beneath the
    * title header and description, then include that content inside the
    * directives. Example:
    *
-   * <hz-page-header header="'My Header' | translate" description="'foo' | translate">
+   * <hz-page-header header="{$ 'My Header' | translate $}" description="{$ 'foo' | translate $}">
    *   <a href="http://www.openstack.org">Go to OpenStack</a>
    * </hz-page-header>
    */
@@ -49,8 +49,8 @@
     var directive = {
       restrict: 'E',
       scope: {
-        header: '=',
-        description: '='
+        header: '@',
+        description: '@'
       },
       templateUrl: basePath + 'hz-page-header.html',
       transclude: true
diff --git a/horizon/static/framework/widgets/headers/hz-page-header.directive.spec.js b/horizon/static/framework/widgets/headers/hz-page-header.directive.spec.js
index f30b83c..fde9485 100644
--- a/horizon/static/framework/widgets/headers/hz-page-header.directive.spec.js
+++ b/horizon/static/framework/widgets/headers/hz-page-header.directive.spec.js
@@ -36,10 +36,8 @@
     }));
 
     it('defines basic elements', function () {
-      $scope.scopeTitle = 'A Title';
-      $scope.scopeDesc = 'A Description';
-      var markup = '<hz-page-header header="scopeTitle" ' +
-        'description="scopeDesc"><a href="#">A href</a></hz-page-header>';
+      var markup = '<hz-page-header header="A Title" ' +
+        'description="A Description"><a href="#">A href</a></hz-page-header>';
 
       var $element = digestMarkup($scope, $compile, markup);
 
diff --git a/openstack_dashboard/dashboards/identity/ngusers/templates/ngusers/index.html b/openstack_dashboard/dashboards/identity/ngusers/templates/ngusers/index.html
index 0f14199..f0518e3 100644
--- a/openstack_dashboard/dashboards/identity/ngusers/templates/ngusers/index.html
+++ b/openstack_dashboard/dashboards/identity/ngusers/templates/ngusers/index.html
@@ -2,10 +2,6 @@
 {% load i18n %}
 {% block title %}{% trans "Users" %}{% endblock %}
 
-{% block page_header %}
-  {% include "horizon/common/_domain_page_header.html" with title=_("Users") %}
-{% endblock page_header %}
-
 {% block main %}
   <ng-include src="'{{ STATIC_URL }}dashboard/identity/users/table/table.html'"></ng-include>
 {% endblock %}
diff --git a/openstack_dashboard/dashboards/identity/static/dashboard/identity/users/table/table.html b/openstack_dashboard/dashboards/identity/static/dashboard/identity/users/table/table.html
index fcd465e..04ccbe2 100644
--- a/openstack_dashboard/dashboards/identity/static/dashboard/identity/users/table/table.html
+++ b/openstack_dashboard/dashboards/identity/static/dashboard/identity/users/table/table.html
@@ -1,3 +1,5 @@
+<hz-page-header header="{$ 'Users' | translate $}"></hz-page-header>
+
 <table ng-controller="identityUsersTableController as table"
   hz-table ng-cloak
   st-table="table.iusers"
diff --git a/openstack_dashboard/dashboards/project/ngimages/templates/ngimages/index.html b/openstack_dashboard/dashboards/project/ngimages/templates/ngimages/index.html
index 13224a0..2b84cac 100644
--- a/openstack_dashboard/dashboards/project/ngimages/templates/ngimages/index.html
+++ b/openstack_dashboard/dashboards/project/ngimages/templates/ngimages/index.html
@@ -2,10 +2,6 @@
 {% load i18n %}
 {% block title %}{% trans "Images" %}{% endblock %}
 
-{% block page_header %}
-  <hz-page-header header="'{% trans "Images" %}'"></hz-page-header>
-{% endblock page_header %}
-
 {% block main %}
   <ng-include src="'{{ STATIC_URL }}app/core/images/table/images-table.html'"></ng-include>
 {% endblock %}
diff --git a/openstack_dashboard/static/app/core/images/table/images-table.html b/openstack_dashboard/static/app/core/images/table/images-table.html
index 4684d38..7989b6b 100644
--- a/openstack_dashboard/static/app/core/images/table/images-table.html
+++ b/openstack_dashboard/static/app/core/images/table/images-table.html
@@ -1,3 +1,5 @@
+<hz-page-header header="{$ 'Images' | translate $}"></hz-page-header>
+
 <table ng-controller="imagesTableController as table"
        hz-table ng-cloak
        st-table="table.images"
-- 
1.9.1

